x/supplier/keeper/session.go

```diff
@@ -0,0 +1,46 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
+)
+
+func (k msgServer) ValidateSessionHeader(
```

Comment:
1. Do we have unit tests for this? If not, let's add them.
2. Missing godoc comment.
3. `Validate` is often using for basic metadata checks, so Consider renaming `ValidateSessionHeaderAgainstChain` or `VerifySessionHeaderIntegrity` or `VerifySessionHeaderCorrectness` to imply it's doing a call. The godoc might help with it too.

Reply:
> 1. Do we have unit tests for this? If not, let's add them.

This is covered by `msg_server_create_claim_test.go` as well as `msg_server_submit_proof.go`.

> 2.  Missing godoc comment.

:+1: 

> 3. Validate is often using for basic metadata checks, so Consider renaming ValidateSessionHeaderAgainstChain or VerifySessionHeaderIntegrity or VerifySessionHeaderCorrectness to imply it's doing a call. The godoc might help with it too.

:thinking:

Reply:
Wdyt about `queryAndValidateSessionHeader()`?

Reply:
I think it's greeeat 🐯